### PR TITLE
[qtbase] Make feature pcre2 to hard dependency when install static library

### LINKED ISSUE #22482: build failure: qtbase[core,gui,doubleconversion]:x64-windows-static
**Host Environment**
- OS: Windows 10 2004
- Compiler: MSVC

**To Reproduce**
`./vcpkg install qtbase[core,gui,doubleconversion]:x64-windows-static`

**Failure logs** 
```
-- The following packages have been found:

...

-- The following REQUIRED packages have not been found:

 * WrapPCRE2

-- The following OPTIONAL packages have not been found:

...

CMake Error at C:/Program Files/CMake/share/cmake-3.22/Modules/FeatureSummary.cmake:464 (message):
  feature_summary() Error: REQUIRED package(s) are missing, aborting CMake
  run.
```

**Additional context**

Adding `pcre2` feature fixes the problem: `qtbase[core,gui,doubleconversion,pcre2]:x64-windows-static`.

I think it would be great if pcre2 is added automatically when needed.

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #22482 
When install `qtbase[core,gui,doubleconversion]:x64-windows-static`, `qtbase` will build failed with the following error:
```
-- The following REQUIRED packages have not been found:
 * WrapPCRE2
-- The following OPTIONAL packages have not been found:
...
CMake Error at C:/Program Files/CMake/share/cmake-3.22/Modules/FeatureSummary.cmake:464 (message):
  feature_summary() Error: REQUIRED package(s) are missing, aborting CMake
  run.
```
This need install `qtbase[pcre2]`, add fatal error to remind install feature `pcre2`.

### CODE REVIEW

File: ports/qtbase/portfile.cmake [hunk 1]
```diff
@@ -24,8 +24,13 @@ if(NOT VCPKG_USE_HEAD_VERSION AND NOT QT_IS_LATEST)
...
 
-if(VCPKG_TARGET_IS_WINDOWS AND NOT "doubleconversion" IN_LIST FEATURES)
-    message(FATAL_ERROR "${PORT} requires feature doubleconversion on windows!" )
+if(VCPKG_TARGET_IS_WINDOWS)
+    if(NOT "doubleconversion" IN_LIST FEATURES)
+        message(FATAL_ERROR "${PORT} requires feature doubleconversion on windows!" )
+    endif()
+    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" AND NOT "pcre2" IN_LIST FEATURES)
+        message(FATAL_ERROR "${PORT} requires feature pcre2 on windows!" )
+    endif()
```
Comment (Neumann-A) [inline ports/qtbase/portfile.cmake:33]: No... make it a hard dependency instead -> see #22012

File: ports/qtbase/vcpkg.json [hunk 2]
```diff
@@ -16,6 +16,14 @@
       "host": true,
       "default-features": false
     },
+    {
+      "name": "qtbase",
+      "default-features": false,
+      "features": [
+        "pcre2"
+      ],
+      "platform": "windows"
```
Comment (Neumann-A) [inline ports/qtbase/vcpkg.json:25]:
```suggestion
      "platform": "windows & static"
```
Your initial commit only did this for static builds? Are dynamic builds also affected?

Please also do this for double-conversion (from #22012) on windows, so the portfile is not changed twice just for that.

Comment (LilyWangLL) [reply ports/qtbase/vcpkg.json:25]: Due to I test dynamic build faild locally, I didn't add `static`. I double check dynamic builds, that faild reason is not related with feature `pcre2`. 
Done, thanks a lot.

File: ports/qtbase/vcpkg.json [hunk 3]
```diff
@@ -16,6 +16,14 @@
...
       "default-features": false
     },
+    {
+      "name": "qtbase",
+      "default-features": false,
+      "features": [
+        "pcre2"
+      ],
+      "platform": "windows & static"
+    },
```
Comment (Neumann-A) [inline ports/qtbase/vcpkg.json:26]:
```suggestion
    {
      "name": "qtbase",
      "default-features": false,
      "features": [
        "pcre2"
      ],
      "platform": "windows & static"
    },
    {
      "name": "qtbase",
      "default-features": false,
      "features": [
        "doubleconversion"
      ],
      "platform": "windows"
    },
```

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
This LGTM, thanks!

**Pull request merged**